### DISCUSSION TITLE
“Mandatory” fields in MSR?  (=> User Guide update?)

### DISCUSSION THREAD

**Discussion by jhaasdyk-au** [GeoscienceAustralia/DynAdjust#196] [Q&A]
The purpose of this github discussion is to:
- record findings, feedback and questions to inform planned discussions with icsm-au/DynAdjust subject matter experts
   i.e. to inform additional user(self)-education, or potential enhancements or defects.
 
All findings and recommendations are my own and may not correctly represent current or planned DynAdjust behaviour.

---

PRIORITY: LOW

This discussion asks if DynAdjust Users Guide could/should clarify the following:

- Currently Section B.3.2. suggests all attributes listed in the tables in B.1.4 (Measurement types) is mandatory. 
- Section B.1.4 itself doesn’t (in general) indicate what is mandatory / compulsory or not.
- Practice suggests that not all fields listed in Section B.1.4 are mandatory. 
- Also, Section B.1.2 says that the epoch (and reference frame) in file header will be overwritten by the epoch (and reference frame) provided for each GNSS measurement.
Why only for GNSS measurement, when all MSR types support the epoch field?

I would suggest an update to Section B.1.4 to clarify the above.

---
**BACKGROUND:**
_All references below are to the DynAdjust Users Manual (Sept 2022) unless otherwise noted:_

**Section B.3.2 Measurement information (DynAdjust Markup Language (DynaML) format)**
-	“Not all elements [in the schema] are mandatory for each measurement”, but… 
-	**“The mandatory elements and required format for the respective measurement types (c.f. Type) follow the definitions contained in the DNA format specification (see §B.1.4).”**

**Section 3.2.7 Verification and error checking.**
-	“Refer to the file format specification in Appendix B [for a list of mandatory fields and elements…”

**Section B.1.2 Header line**
-	(This is interesting, but not surprising [No change suggested here])
“The reference frame/geoid model and epoch/version fields [in the header line] are non–mandatory”

**Section 3.3.1 Reference Frame**
-	Footnote 1: “Both DynaML and GeodesyML contain elements for managing reference frame and epoch, although this information is not mandatory.”

<br/>
**IN PRACTICE:** <br/>
Section B.1.4 Measurement information (including subsections for each measurement type)

- Not every field listed per measurement type appears to be mandatory. For example:  
   - Epoch is optional: Epoch is listed for all MSR types, but most examples exclude epoch. 
        [Note: epoch can be (optionally) also provided in file header]
   - Reference frame is optional.  Not all examples (of GNSS type msr) include reference frame (e.g. Figure B.7 GNSS Baseline).
        [Note: reference frame can be (optionally) also provided in file header]
   - Q) Are all of Degrees, Minutes and Seconds actually mandatory? What is the actual behaviour of import.exe if missing?
   - Q) Is standard deviation mandatory? What is the actual behaviour of import.exe if missing?

It is also interesting to note:
-	Only GNSS-type MSR support the reference frame attribute. [See separate discussion on this point (link to come)]
-	Table B.2 indicates that a DNA station file can NOT assign epoch to individual stations lines. Is there scope for an enhancement here?
-	Section B.1.2 “For measurement files, the default reference frame and epoch [that is, default for the file, as provided in the header] will be overridden by the reference frame and epoch provided for each GNSS measurement”. 
     -	Since all MSR types can have an epoch…
     -	is this behaviour actually limited to GNSS types?
     -	See related discusion [Should 'reference frame' (and epoch) apply to more than just GNSS-type measurements?](https://github.com/icsm-au/DynAdjust/discussions/198#discussion-4376793)

**Comment by rogerfraser** [comment]
Thanks @jhaasdyk-au.  Yes, clarification on mandatory fields should be added to the documentation.  This can be accommodated in the next release.
